dev/system-test/firestore.ts

```diff
@@ -1247,54 +1248,42 @@ describe('Query class', () => {
       });
   });
 
-  it('has limit() method', () => {
-    const ref1 = randomCol.doc('doc1');
-    const ref2 = randomCol.doc('doc2');
-
-    return Promise.all([ref1.set({foo: 'a'}), ref2.set({foo: 'b'})])
-      .then(() => {
-        return randomCol
-          .orderBy('foo')
-          .limit(1)
-          .get();
-      })
-      .then(res => {
-        expect(res.size).to.equal(1);
-        expect(res.docs[0].data()).to.deep.equal({foo: 'a'});
-      });
+  it('has limit() method', async () => {
+    await addDocs({foo: 'a'}, {foo: 'b'});
+    const res = await randomCol
+      .orderBy('foo')
+      .limit(1)
+      .get();
+    expectDocs(res, {foo: 'a'});
   });
 
-  it('has offset() method', () => {
-    const ref1 = randomCol.doc('doc1');
-    const ref2 = randomCol.doc('doc2');
-
-    return Promise.all([ref1.set({foo: 'a'}), ref2.set({foo: 'b'})])
-      .then(() => {
-        return randomCol
-          .orderBy('foo')
-          .offset(1)
-          .get();
-      })
-      .then(res => {
-        expect(res.size).to.equal(1);
-        expect(res.docs[0].data()).to.deep.equal({foo: 'b'});
-      });
+  it('has limitToLast() method', async () => {
+    await addDocs({doc: 1}, {doc: 2}, {doc: 3});
+    const res = await randomCol
+      .orderBy('doc')
+      .limitToLast(2)
+      .get();
+    expectDocs(res, {doc: 2}, {doc: 3});
   });
 
-  it('has startAt() method', () => {
-    const ref1 = randomCol.doc('doc1');
-    const ref2 = randomCol.doc('doc2');
+  it('limitToLast() supports Query cursors', async () => {
```

Comment:
Can we add error cases for `limitToLast()`, like cannot be streamed and requires explicit orderBy?